server/src/main/java/org/opensearch/ingest/IngestService.java

```diff
@@ -107,6 +108,18 @@ public class IngestService implements ClusterStateApplier, ReportingService<Inge
 
     public static final String INGEST_ORIGIN = "ingest";
 
+    /**
+     * Defines the limit for the number of processors which can run on a given document during ingestion.
+     */
+    public static final Setting<Integer> MAX_NUMBER_OF_INGEST_PROCESSORS = Setting.intSetting(
+        "cluster.ingest.max_number_processors",
+        Integer.MAX_VALUE,
```

Comment:
Add a tracking issue to change default to reasonable no. like 50 or so in 3.0 as this would be breaking change

Reply:
@dblock do you have an opinion here? should we add some reasonable default now also may be like 100 and call it in release notes even if it is breaking change.

Reply:
Created this tracking issue : https://github.com/opensearch-project/OpenSearch/issues/15498

Reply:
Hi @dblock,
Let us know your thoughts :)